### DISCUSSION TITLE
[Resolved] Expected computation time for t1-volume pipeline

### DISCUSSION THREAD

**Discussion by AudreyDuran** [aramis-lab/clinica#824] [General]
Hi again clinica team,

Is it possible to get a rough idea of the expected computation time for the different sub-pipeline of  `t1-volume`  (`t1-volume-create-dartel` etc) ? An approximation of the time needed / image for example would be very appreciated.

Thanks !

Audrey

**Comment by AudreyDuran** [comment]
So finally `t1-volume-create-dartel` seems to be the longest part of `t1-volume` pipeline and was completed after 41 hours with 20 processes on 637 patients with 2 sessions / patients (=1274 volumes).

**Reply by NicolasGensollen** [reply]
Hi @AudreyDuran 
Thanks a lot for looking into this and providing this kind of feedback. I don't think we have currently this kind of information available for the different pipelines of Clinica. I suspect that, even normalized by volumes and CPU cores, the running time will depend on a lot of different factors related to the pipeline configuration as well as the executing machine.
I do agree that providing a rough estimate could be a useful piece of information to put in the user documentation though.

**Reply by ghisvail** [reply]
I agree that precise runtime cost estimation will vary from one machine to another.

A possible area of improvement may be to have some form of dynamic estimation, which would project the time for computation over the remaining subject / sessions based on some average over the completed ones.

Similar to the progress window when copying a large volume of files, which gives a rough projection of the time for completion given the estimated write speed to disk.

**Reply by AudreyDuran** [reply]
Yes precise estimation would be difficult, but even a very rough one would be very valuable (I had to idea if it should be more around 1 hour or 1 day). 
Dynamic estimation would be great !
Thanks for taking into consideration the comments :)
